src/libs/Middleware/Pagination.ts

```diff
@@ -106,13 +105,10 @@ const Pagination: Middleware = (requestResponse, request) => {
 
         const newPage = sortedPageItems.map((item) => getItemID(item));
 
-        // Detect if we are at the start of the list. This will always be the case for the initial request with no cursor.
-        // For previous requests we check that no new data is returned. Ideally the server would return that info.
-        if ((type === 'initial' && !cursorID) || (type === 'next' && newPage.length === 1 && newPage.at(0) === cursorID)) {
+        if (response.hasNewerActions === false || (type === 'initial' && !cursorID)) {
```

Comment:
hmmm I'm wondering if it actually should be more simply like this:

```suggestion
        if (response.hasNewerActions === false) {
```

because the back-end should always return `hasNewerActions`, even if `OpenReport` is called without a `cursorID`.

Reply:
When I tested this it did not, that is why I added this condition back.

Reply:
I can't find a reference to `cursorID` in the backend, is that the `reportActionID` we are linking to?

Reply:
Yes, it is only use in the frontend, it is set here https://github.com/Expensify/App/blob/0c4b2d1121139b8e376a1f3d222065a6a4f143d6/src/libs/actions/Report.ts#L973.

Would it be possible to make the backend return both hasNext and hasPrevious even if no cursor is sent, it would simply the code here?

Reply:
Just to make sure, `OpenReport` without a `cursorID` means the user is opening the report at the last action right?

Reply:
Actually, it might not always be the case now, see https://github.com/Expensify/App/issues/35011#issuecomment-2311342272. I think we should make sure that OpenReport always return the pagination info even when not passing a cursor.

Reply:
Hmm this is odd, because of this other comment https://github.com/Expensify/App/issues/41153#issuecomment-2316070239 😄 

Anyway, I am not sure what it means calling `OpenReport` without a `reportActionID`, how can I say whether a report has newer actions if I don't know related to what?

I mean, the code we are talking about implies that if type is `initial` and no `cursorID` then `hasNewerActions`  should be false, right?

Reply:
From what I understand in https://github.com/Expensify/App/issues/35011 we want to open report at the last unread message, so we want to remove the assumption that calling OpenReport without a reportActionID means that there are no newer actions and instead rely on hasNewerActions returned from the backend. However at the moment it seems like that information is not returned.

I think our 2 options for now is:

1. Update the backend to always return hasNewerActions even when called with no reportActionID (I do not have access to the backend so it would need to be done by an Expensify engineer).
2. Just leave the code as is for now and remove the check for type is initial and no cursorID whenever we support opening report at the last unread message.

Reply:
Aaah I finally understand, thanks for explaining, let me check what would need to change in order for 1 work.

Reply:
Ok, I figured what needs to be done, but then the App behaves in a few weird ways.

First, if you open a direct link to the report it will load only the unread page and won't load the newer ones, despite having the `hasNewerActions: true` and the code suggestion above applied

Second, if you have the app open and receive the messages then open the report we call ReadNewestAction before we call OpenReport, then we can't calculate whatever you have unread
<img width="327" alt="image" src="https://github.com/user-attachments/assets/edde8111-8de4-49c9-ac70-435a29e8d8a3">

I think we should merge this PR and figure out this problems in the other issue, what do you think @janicduplessis ?

Reply:
Ok sounds good